src/core/net/nd6.cpp

```diff
@@ -200,6 +200,95 @@ uint8_t RecursiveDnsServerOption::OptionLengthFor(uint8_t aNumAddresses)
     return ClampToUint8(DivideAndRoundUp(size, kLengthUnit));
 }
 
+//----------------------------------------------------------------------------------------------------------------------
+// Nat64PrefixInfoOption
+
+void Nat64PrefixInfoOption::Init(void)
+{
+    Clear();
+    SetType(kTypeNat64PrefixInfo);
+    SetSize(sizeof(Nat64PrefixInfoOption));
+    // Per RFC8781, the length of PREF64 option is in units of 8 octets and MUST be 2.
+    SetLength(2);
+}
+
+void Nat64PrefixInfoOption::SetLifetime(uint16_t aLifetime)
+{
+    uint32_t scaledLifetime = DivideAndRoundUp(aLifetime, kLifetimeScalingUnit);
+
+    mPrefixAttr &= ~kScaledLifetimeMask;
+    mPrefixAttr |= ClampToUint16(scaledLifetime << kScaledLifetimeOffset);
+}
+
+void Nat64PrefixInfoOption::SetPrefixLengthCode(const uint8_t aPrefixLengthCode)
+{
+    mPrefixAttr &= ~kPrefixLengthCodeMask;
+    mPrefixAttr |= (aPrefixLengthCode & kPrefixLengthCodeMask);
+}
+
+Error Nat64PrefixInfoOption::SetPrefix(const Prefix &aPrefix)
+{
+    Error error = kErrorNone;
+
+    memcpy(mPrefixMsb, aPrefix.GetBytes(), 12);
+
+    switch (aPrefix.mLength)
+    {
+    case 96:
+        SetPrefixLengthCode(0);
+        break;
+    case 64:
+        SetPrefixLengthCode(1);
+        break;
+    case 56:
+        SetPrefixLengthCode(2);
+        break;
+    case 48:
+        SetPrefixLengthCode(3);
+        break;
+    case 40:
+        SetPrefixLengthCode(4);
+        break;
+    case 32:
+        SetPrefixLengthCode(5);
+        break;
+    default:
+        error = kErrorInvalidArgs;
+    }
+
+    return error;
+}
+
+void Nat64PrefixInfoOption::GetPrefix(Prefix &aPrefix) const
+{
+    uint16_t prefixLength;
+
+    switch (GetPrefixLengthCode())
+    {
+    case 1:
+        prefixLength = 64;
+        break;
+    case 2:
+        prefixLength = 56;
+        break;
+    case 3:
+        prefixLength = 48;
+        break;
+    case 4:
+        prefixLength = 40;
+        break;
+    case 5:
+        prefixLength = 32;
+        break;
+    default:
+        prefixLength = 96;
```

Comment:
Can consider defining an array for mapping from code to prefix length? We can define it as `const` in the `Nat64PrefixInfoOption` class, so we can use it from different methods?

We can also use the array in `SetPrefix()` (i.e., iterate and search in the short array to find a match and determine the associated length-code).

What do you think?